draft-ietf-dnsop-svcb-https.md

```diff
@@ -652,10 +652,12 @@ See {{incomplete-response}} for possible optimizations of this procedure.
 ## General requirements
 
 Recursive resolvers MUST be able to convey SVCB records with unrecognized
-SvcParamKeys or malformed SvcParamValues.  Resolvers MAY treat the entire
+SvcParamKeys, and MAY treat the entire
 SvcParams portion of the record as opaque.  No part of this specification requires
 recursive resolvers to alter their behavior based on its contents, even if the contents
-are invalid.
+are invalid.  To ensure compatibility with complex SvcParam specifications,
```

Comment:
```suggestion
are invalid.  To ensure compatibility with complex and/or future SvcParam specifications,
```

Reply:
Why isn't this covered by the preceding requirement to pass unrecognized keys?

Reply:
I guess it mostly is, but if we're adding justification does it hurt to remind people over and over?

Reply:
I don't think it's relevant to this sentence, unless you're imagining changes that formally alter the range of allowed values for an existing key.

Reply:
An example might be future alpn values.  Perhaps there's a better way to work?

Reply:
"evolving"?

Reply:
This spec identifies ALPN IDs as "a sequence of 1-255 octets", and in no way implies that anyone should restrict validation to currently defined identifiers.  If some future ALPN ID is more than 255 octets, there are larger problems, in that the existing parameter format wouldn't be able to encode them.

Reply:
(TLS uses a 1-byte length prefix, so it's limited to 255 octets there too.)